### ISSUE TITLE
[ray] 0.8.5 confused error message when starting driver node soon after creating the cluster

### ISSUE THREAD

**Issue by nlnjnj** [ray-project/ray#9170]
I got the following error message when starting driver node soon after creating the cluster on k8s.

`ray start --node-ip-address=$MY_POD_IP --num-cpus=0 --address=$RAY_HEAD_SERVICE_HOST:$RAY_HEAD_SERVICE_PORT_REDIS_PRIMARY --object-manager-port=12345 --internal-config="{\"raylet_heartbeat_timeout_milliseconds\":5000}" --node-manager-port=12346 && python script.py`

```
2020-06-27 15:15:58,208	INFO scripts.py:429 -- Using IP address 10.244.39.210 for this node.
Traceback (most recent call last):
  File "/opt/conda/bin/ray", line 8, in <module>
    sys.exit(main())
  File "/opt/conda/lib/python3.7/site-packages/ray/scripts/scripts.py", line 1028, in main
    return cli()
  File "/opt/conda/lib/python3.7/site-packages/click/core.py", line 829, in __call__
    return self.main(*args, **kwargs)
  File "/opt/conda/lib/python3.7/site-packages/click/core.py", line 782, in main
    rv = self.invoke(ctx)
  File "/opt/conda/lib/python3.7/site-packages/click/core.py", line 1259, in invoke
    return _process_result(sub_ctx.command.invoke(sub_ctx))
  File "/opt/conda/lib/python3.7/site-packages/click/core.py", line 1066, in invoke
    return ctx.invoke(self.callback, **ctx.params)
  File "/opt/conda/lib/python3.7/site-packages/click/core.py", line 610, in invoke
    return callback(*args, **kwargs)
  File "/opt/conda/lib/python3.7/site-packages/ray/scripts/scripts.py", line 437, in start
    ray_params, head=False, shutdown_at_exit=block, spawn_reaper=block)
  File "/opt/conda/lib/python3.7/site-packages/ray/node.py", line 120, in __init__
    redis_client.get("session_name"))
  File "/opt/conda/lib/python3.7/site-packages/ray/utils.py", line 216, in decode
    "The argument {} must be a bytes object.".format(byte_str))
ValueError: The argument None must be a bytes object.
```

So I the lookup the corresponding source code trying to figure it out, the reason seems to be the None value when `redis_client.get("session_name"))`.

ray/node.py", line 120
```python
        if head:
            redis_client = None
            # date including microsecond
            date_str = datetime.datetime.today().strftime(
                "%Y-%m-%d_%H-%M-%S_%f")
            self.session_name = "session_{date_str}_{pid}".format(
                pid=os.getpid(), date_str=date_str)
        else:
            redis_client = self.create_redis_client()
            self.session_name = ray.utils.decode(
                redis_client.get("session_name"))
```

In summary, my issue is that should we present the correct error message like 'ray head is not started yet', or should we implement a headless cluster strategy to avoid this issue.

**Comment by richardliaw** [comment]
@nlnjnj this should be fixed on the current master.

**Comment by richardliaw** [comment]
This is a duplicate of https://github.com/ray-project/ray/issues/5119
